### PR TITLE
simple iterator implementation

### PR DESCRIPTION
**Pull request by theSherwood**
Thanks for the interval tree implementation. I found I need some kind of cursor implementation. This is WIP, but I wanted to get it on your radar. Is a cursor implementation something you're open to?

### CODE REVIEW

**Comment by theSherwood** [comment]
I've removed the bi-directionality (`prev` method on the cursor). Due to the bias of the sorting method, navigating backwards yields potentially surprising results. It would also be the only case of any api in the codebase that navigates backwards. So I removed it.

**Comment by alexbol99** [comment]
Hi @theSherwood ,
This is a great contribution to the project.
But isn't better to call this mechanism "iterator"? 
AFAIK "cursor" is typically associated with databases, where it represents the current position in a result set, not a traverse mechanism itself.
What do you think?

**Comment by theSherwood** [comment]
@alexbol99 No problem! I'll make those changes and get it pushed up.

**Comment by alexbol99** [comment]
@theSherwood ,
So maybe you can take it even one step further and implement it as iterable so that one can use `for .. of` loop?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators

**Comment by theSherwood** [comment]
@alexbol99 Sure. I can do that

**Comment by theSherwood** [comment]
@alexbol99 Good call. Using a generator function made things a lot simpler.

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
   "types": "index.d.ts",
   "nyc": {
     "require": [
-      "@babel/register"
+      "@babel/register",
+      "@babel/polyfill"
```
Comment (theSherwood) [inline package.json:15]: This line is needed for the tests to be able to handle the generator function.

**Comment by alexbol99** [comment]
Looks great!
I will merge and release it tomorrow
Thank you so much for collaboration

**Comment by theSherwood** [comment]
Great! Thanks for the quick turnaround on it all

**Pull request merged**